nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/PublishGCPubSub.java

```diff
@@ -137,9 +211,10 @@ public Set<Relationship> getRelationships() {
     @OnScheduled
     public void onScheduled(ProcessContext context) {
         try {
+            storedException.set(null);
             publisher = getPublisherBuilder(context).build();
         } catch (IOException e) {
-            getLogger().error("Failed to create Google Cloud PubSub Publisher due to {}", new Object[]{e});
+            getLogger().error("Failed to create Google Cloud PubSub Publisher due to {}", e);

```

Comment:
The `due to {}` approach should not be used because the logger writes the entire stack trace from the exception. This error logging should be retained and the exception can be thrown after logging.
```suggestion
            getLogger().error("Failed to create Google Cloud PubSub Publisher", e);

```

Reply:
I understand that an exception throw would result in the exception being logged, so going with a one-liner for clarity.